src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayController.cs

```diff
@@ -139,62 +136,57 @@ public async Task<IActionResult> Evaluated(Guid applicationId)
+                    SetupGatewayViewModelErrorMessagesAndValues(model, vm);
+                    return View("~/Views/Roatp/Apply/Gateway/pages/LegalName.cshtml", model);
             }
-            
 
-            // this is temporary, the important thing is to save, including user name, then redirect back to overview
-            var model = new RoatpGatewayPageViewModel { ApplicationId = vm.ApplicationId, PageId = vm.PageId, Value=vm.Value, OptionPassText = vm.OptionPassText, OptionFailText = vm.OptionFailText, OptionInProgressText = vm.OptionInProgressText};
-            model.NextPageId = vm.PageId;
+            vm.SourcesCheckedOn = DateTime.Now;
 
-            // if it gets here, save it....
+            var pageData = JsonConvert.SerializeObject(vm);
 
+            var username = _contextAccessor.HttpContext.User.UserDisplayName();
+            await _applyApiClient.SubmitGatewayPageAnswer(vm.ApplicationId, vm.PageId, vm.Status, username, pageData);
 
-
-            // go to overview page
-            //return RedirectToAction("GetGatewayPage", new { model.ApplicationId, pageId = model.NextPageId });
             return RedirectToAction("ViewApplication", new {vm.ApplicationId});
 
         }
 
-
-        [HttpGet("/Roatp/Gateway/{applicationId}/Page/{PageId}")]
-        public async Task<IActionResult> GetGatewayPage(Guid applicationId, string pageId)
+        [HttpGet("/Roatp/Gateway/{applicationId}/Page/1-10")]
+        public async Task<IActionResult> GetGatewayLegalNamePage(Guid applicationId, string pageId)
         {
-           var model = new RoatpGatewayPageViewModel {ApplicationId = applicationId, PageId = "NotFound"};
-
-
-           if (pageId == "1-10")
-           {
-               model = await _mediator.Send(new GetLegalNameRequest(applicationId));
-           }
+            return View("~/Views/Roatp/Apply/Gateway/pages/LegalName.cshtml", await _mediator.Send(new GetLegalNameRequest(applicationId)));
+        }
 
-           if (model.PageId == "NotFound")
-           {
-               return View("~/Views/ErrorPage/PageNotFound.cshtml");
-            }
+        private static void SetupGatewayViewModelErrorMessagesAndValues(LegalNamePageViewModel model, LegalNamePageViewModel vm)
+        {
+            model.ErrorMessages = vm.ErrorMessages;
+            model.Status = vm.Status;
+            model.OptionInProgressText = vm.OptionInProgressText;
+            model.OptionFailText = vm.OptionFailText;
+            model.OptionPassText = vm.OptionPassText;
+        }
 
-           return View("~/Views/Roatp/Apply/Gateway/Page.cshtml", model);
+        private static void SetupGatewayPageOptionTexts(LegalNamePageViewModel vm)
+        {
+            if (vm?.Status == null) return;
+            vm.OptionInProgressText = vm.Status.ToLower() == "in progress" && !string.IsNullOrEmpty(vm.OptionInProgressText) ? vm.OptionInProgressText : string.Empty;
```

Comment:
Could do with converting these magic strings into enums

Reply:
will do in next story